packages/jamf_protect/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,468 @@
+                    ctx.process.parent.code_signature.team_id = parentProcess.signingInfo.teamid;
+                }
+
+                }
+
+                // Mapping out the process group leader, which can be the same as parent
+                def processGroupLeader = ctx.jamf_protect.alerts.input.related.processes[ctx.jamf_protect.alerts.input.related.processes.size() - 1];
+                ctx.process.group_leader = new HashMap();
+                ctx.process.group_leader.name = processGroupLeader.name;
+                ctx.process.group_leader.pid = processGroupLeader.pid;
+                ctx.process.group_leader.executable = processGroupLeader.path;
+
+                if (processGroupLeader.containsKey('startTimestamp')) {
+                    ctx.process.group_leader.start = Instant.ofEpochSecond(processGroupLeader.startTimestamp).toString();
+                }
+
+                ctx.process.group_leader.user = new HashMap();
+                if (processGroupLeader?.uid != null) {
+                    ctx.process.group_leader.user.id = processGroupLeader.uid.toString();
+                }      
+
+                ctx.process.group_leader.real_user = new HashMap();
+                if (processGroupLeader?.ruid != null) {
+                    ctx.process.group_leader.real_user.id = processGroupLeader.ruid.toString();
+                }               
+
+                ctx.process.group_leader.real_group = new HashMap();
+                if (processGroupLeader?.rgid != null) {
+                    ctx.process.group_leader.real_group.id = processGroupLeader.rgid.toString();
+                }
+            }
+
+######################
+## ECS Group Mapping ##
+######################
+    - script:
+        description: One scripting processor to capture all the related group information for the event.
+        lang: painless
+        source: >
+            if (ctx.jamf_protect?.alerts?.input?.related?.groups != null && ctx.jamf_protect.alerts.input.related.groups.size() > 0) {
+                def group = ctx.jamf_protect.alerts.input.related.groups[0];
+                
+                    ctx.group = ctx.group ?: new HashMap();
+                    
+                    ctx.group.name = group.name;
+                    ctx.group.id = group.gid.toString();
+            }
+
+########################
+# ECS Threat Mapping ##
+########################
+    - set:
+        field: threat.framework
+        value: "MITRE ATT&CK"
+        if: ctx.jamf_protect?.alerts?.input?.match?.facts[0].tags.contains("MITREattack")
+    - set:
+        field: threat.software.platforms
+        value: macOS
+        if: ctx.jamf_protect?.alerts?.input?.match?.facts[0].tags.contains("MITREattack")
+
+########################
+# ECS Related Mapping ##
+########################
+    - append:
+        field: related.hosts
+        value: "{{ host.name }}"
+        if: ctx?.jamf_protect?.alerts?.input?.host?.hostname != null
```

```suggestion
        if: ctx.host?.hostname != null
```

Better to check directly on the field rather than an implied field. Also for other processors below.